### PR TITLE
ddns-scripts: better error management for gandi

### PR DESCRIPTION
**Pull request by LecrisUT**
Maintainer: @feckert / @Ansuel 

Description:
Added error checks similar to the ones used in `ddns-scripts-pdns` (#16429)

### CODE REVIEW

**Comment by feckert** [comment]
Please remove the https://github.com/openwrt/packages/pull/16430/commits/4e9172f3fd3b61fe3b95c84a54c34debffdf51a the readybility script.
If you want to improve this then use `. /usr/share/libubox/jshn.sh` to build the json. You can then add that with json_dump.
See for example https://github.com/openwrt/openwrt/blob/ec6293febc244d187e71a6e54f44920be679cde4/package/network/utils/comgt/files/directip.sh#L93

**Comment by LecrisUT** [comment]
Hopefully that the right approach, I'll test it once I'm at the router.

File: net/ddns-scripts/files/usr/lib/ddns/update_gandi_net.sh [hunk 1]
```diff
@@ -1,19 +1,37 @@
...
 [ -z "$password" ] && write_log 14 "Service section not configured correctly! Missing API Key as 'password'"
 
 [ $use_ipv6 -ne 0 ] && __RRTYPE="AAAA" || __RRTYPE="A"
 
-curl -s -X PUT "$__ENDPOINT/domains/$domain/records/$username/$__RRTYPE" \
+# Construct JSON payload
+json_init
+json_add_string rrset_ttl "$__TTL"
+json_add_array rrset_values "$__IP"
+ubus call network add_dynamic "$(json_dump)"
```
Comment (feckert) [inline net/ddns-scripts/files/usr/lib/ddns/update_gandi_net.sh:20]: You could delete this line.
This is not necessary.
I just wanted to show you where you can have a look how to use jshn.

**Review by feckert — CHANGES_REQUESTED** [review]

File: net/ddns-scripts/files/usr/lib/ddns/update_gandi_net.sh (hunk 1 above)
Comment (LecrisUT) [reply net/ddns-scripts/files/usr/lib/ddns/update_gandi_net.sh:20]: Oh, my bad, I thought I've deleted that part

Comment (feckert) [reply net/ddns-scripts/files/usr/lib/ddns/update_gandi_net.sh]: Please stash your commits to one after I have merged https://github.com/openwrt/packages/pull/16429 you should the bump the PKG_RELEASE number

Comment (LecrisUT) [reply net/ddns-scripts/files/usr/lib/ddns/update_gandi_net.sh]: Certainly, I will test these changes for myself as well and rebase all of the commits.
So the `PKG_RELEASE` should be bumped for both?

Comment (feckert) [reply net/ddns-scripts/files/usr/lib/ddns/update_gandi_net.sh]: > Certainly, I will test these changes for myself as well and rebase all of the commits.
> So the `PKG_RELEASE` should be bumped for both?
 
Yes we need a PKG_RELEASE bump for this pullrequest and for the other one.

> 
> Edit: BTW, what's with the CI error for `x86_64`?

I have asked myself the same question. I'll have to ask the others. I haven't looked at it yet.

**Comment by LecrisUT** [comment]
Rebased this one on the `PowerDNS` one so we don't get a conflict later on. All changes are squashed now and everything has been tested.

I've also switched it to `LiveDNS API` since the old API has been/is being deprecated from conversations with Gandi team.

**Pull request merged**